src/github_utils.py

```diff
@@ -19,111 +20,107 @@ def get_pull_request_diff():
     # Get diff content
     diff_url = pull_request.diff_url
     response = requests.get(diff_url)
-    print(response.text)
+    #print(response.text)
     return response.text
 
 
-def get_valid_lines(diff_content):
-    """Parse diff to find valid line numbers with proper initialization"""
-    valid_lines = {}
+def get_valid_hunks(diff_content: str) -> List[dict]:
+    """Parse diff content with precise line tracking"""
+    hunks = []
     current_file = None
```

Comment:
The original implementation of this function had a flawed logic for tracking line numbers. This version implements a more robust and accurate hunk parsing method. The old method reset the line numbers for each file, making line numbers inconsistent.
```suggestion
|
```
N/A